yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -395,6 +395,10 @@ private[yarn] class YarnAllocator(
       val executorId = executorIdCounter.toString
 
       assert(container.getResource.getMemory >= resource.getMemory)
```

Comment:
True, I mean, should we expect there is a case where the granted memory is less than requested as well? and allow or handle it? right now it's rejected, so I expect it can't happen. But then again the code seemed to assume that (sort of) about vcores too.